packages/repository/src/relations/relation.helpers.ts

```diff
@@ -60,9 +54,26 @@ export async function findByForeignKeys<
   }
 
   const where = ({[fkName]: value} as unknown) as Where<Target>;
-  const targetFilter = {where};
+  if (scope && !_.isEmpty(scope)) {
+    scope = await combineFilters(where, scope);
+  } else {
+    scope = {where} as Filter<Target>;
+  }
+
+  return targetRepository.find(scope, options);
+}
 
-  return targetRepository.find(targetFilter, options);
+/** A helper that combines where clauses for the scope filter
+ *
+ * @param whereClause a new added where clause
+ * @param scope the passed in scope filter
+ */
+async function combineFilters<Target extends Entity>(
+  whereClause: Where<Target>,
+  scope: Filter<Target>,
+): Promise<Filter<Target>> {
+  scope.where = Object.assign(whereClause, scope.where);
```

Comment:
I believe we have utility functions to combine multiple where clauses safely, to ensure operators like `and` and `or` are handled correctly. @raymondfeng do you remember where to find these helpers? I think it's something like `FilterBuilder` and `WhereBuilder`.

Reply:
Applied